input_manager/include/core/EditorInputManager.hpp

```diff
@@ -8,25 +8,170 @@
 
 #include <data_types/rendering/RenderingContext.hpp>
 
+#include <data_types/math/euler_angles.hpp>
+
 namespace rcbe::core {
 class EditorInputManager final : protected InputManagerImplementation {
+    using input_event_reference = InputManagerImplementation::input_event_reference;
+    using previous_event_reference = InputManagerImplementation::previous_event_reference;
 public:
-    EditorInputManager() = delete;
-    explicit EditorInputManager(const rcbe::rendering::RenderingContextPtr& rendering_context);
+    using handler_collection = std::vector<InputManagerImplementation::handler_intermidiate_storage>;
+
+    EditorInputManager() = default;
+    explicit EditorInputManager(handler_collection&& h);
     ~EditorInputManager() = default;
     EditorInputManager(const EditorInputManager& other) = delete;
     EditorInputManager(EditorInputManager&& other) = default;
     EditorInputManager& operator=(const EditorInputManager& other) = delete;
     EditorInputManager& operator=(EditorInputManager&& other) = default;
 
-    bool try_process_event(InputManagerImplementation::input_event_reference event);
+    // no pack expansions for lambda capture in c++17 so waiting on c++20
+    // until then only one object for capture here
+    template <typename Capture>
+    static EditorInputManager create(const rendering::RenderingContextPtr& ctx, const Capture& c) {
```

Comment:
Seems template is used to avoid including the header, since otherwise it has no reason to do it. If it's the case why not make context a template parameter too?